pkg/klusterlet/clusterinfo/clusterinfo_controller.go

```diff
@@ -531,24 +534,53 @@ func (r *ClusterInfoReconciler) getDistributionInfo() (clusterv1beta1.Distributi
 				continue
 			}
 			if conditionstatus == "True" {
-				upgradeFailed = true
+				ocpDistributionInfo.UpgradeFailed = true
 			}
 			break
 		}
 	}
 
-	distributionInfo.OCP = clusterv1beta1.OCPDistributionInfo{
-		Version: version,
-		// status.desired.version
-		DesiredVersion: desiredVersion,
-		// true when the first status.conditions element with type === Failing has status === True
-		UpgradeFailed: upgradeFailed,
-		// the value of the version property from each element in status.availableUpdates
-		AvailableUpdates: availableVersions,
+	return ocpDistributionInfo, nil
+}
+
+func (r *ClusterInfoReconciler) getOCPClusterID() (string, error) {
```

Comment:
seems to me that getOCPDistributionInfo already gives us the clusterID. How about we also return cluster ID from getOCPDistributionInfo?

Reply:
done. using  `getOCPDistributionInfo` can reduce one API call.